content/code-signing/ios-code-signing.md

```diff
@@ -34,45 +34,63 @@ For successful signing, the certificate and the provisioning profile must match
-3. Click **Save**. Codemagic will attempt to establish a connection to Apple Developer Portal and will ask for a verification code for two-factor authentication or two-step verification. 
-    >If you have set up several trusted phone numbers, select the phone number to which the verification code will be sent.
-4. Enter the verification code and click **Save** one more time. On successful authentication, the Apple Developer Portal integration will be enabled.
+1. Log in to App Store Connect and navigate to **Users and Access > Keys**.
+2. Click on the + sign to generate a new API key.
+3. Enter the name for the key and select an access level. We recommend choosing either `Developer` or `App Manager`, read more about Apple Developer Program role permissions [here](https://help.apple.com/app-store-connect/#/deve5f9a89d7).
+4. Click **Generate**.
+5. As soon as the key is generated, you can see it added in the list of active keys. Click **Download API Key** to save the private key. Note that the key can only be downloaded once.
 
-### Setting up automatic code signing
+{{<notebox >}} 
+Take note of the **Issuer ID** above the table of active keys as well as the **Key ID** of the generated key as these will be required when setting up the Apple Developer Portal integration in Codemagic UI.
+{{</notebox>}}
+
+### Step 2. Connecting the Apple Developer Portal integration for your team/account
+
+The Apple Developer Portal integration can be enabled in **User settings > Integrations** for personal projects and in **Team settings > Team integrations** for projects shared in the team (if you're the team owner). This allows you to conveniently use the same credentials for automatic code signing across different apps and workflows.
 
 {{<notebox >}} 
-The actions that Codemagic can perform on your behalf depend on your [Apple Developer Program user role](https://help.apple.com/app-store-connect/#/deve5f9a89d7).
+Note that users how had previously set up the session-based intgeration have been automatically migrated to use the API-key based setup.
 {{</notebox>}}
 
+1. In the list of available integrations, click the **Connect** button for **Developer Portal**.
+2. In the **App Store Connect API key name**, provide a name for the key you are going to set up the integration with. This is for identifying the key in Codemagic.
+3. Enter the **Issuer ID** related to your Apple Developer account. You can find it above the table of active keys on the Keys tab of the [Users and Access](https://appstoreconnect.apple.com/access/api) page.
+4. Enter the **Key ID** of the key to be used for code signing.
+5. In the **API key** field, upload a RSA 2048 bit private key to be included in the [signing certificate](https://help.apple.com/xcode/mac/current/#/dev1c7c2c67d) that Codemagic creates. You can use an existing key or create a new 2048 bit RSA key by running the following command in your terminal:
```

Comment:
This is not correct. API key is the private key that is downloaded from App Store Connect. The thingie that can be downloaded only once after it is generated.

Reply:
Updated. 
FYI, I took it from our yaml documentation - do we need to revise there as well?

Reply:
I think so.